### PR TITLE
docs: fixing upgrading guide v2

### LINKED ISSUE #153: Incorrect rendering of tables in the docs
In the [generated docs](https://googleapis.dev/python/datacatalog/latest/UPGRADING.html#supported-python-versions), the tables containing "applicable previous versions" are not rendered correctly, while in the source markdown file [UPGRADING.md](https://github.com/googleapis/python-datacatalog/blob/bb2765dc219580d91ff82611971631b1a62bd41e/UPGRADING.md#supported-python-versions) they are rendered just fine.

### PR DESCRIPTION
**Pull request by meredithslota**
Removed formatted tables (reverts #114)
Converted file back to .MD (reverts part of #268)
Fix link to file

Fixes #153  🦕

### CODE REVIEW

**Comment by meredithslota** [comment]
I do not totally understand why the Owlbot changes got bundled into this PR.

**Review by dandhlee — APPROVED** [review]
Could be that owlbot updates unfroze and trying to put updates in. LGTM for just formatting update though.

**Comment by meredithslota** [comment]
Owlbot changes were merged in a seperate PR so we are back down to just formatting changes. :)

File: docs/index.rst [hunk 1]
```diff
@@ -20,7 +20,18 @@ API Reference
     :maxdepth: 2
 
     datacatalog_v1beta1/services
-    datacatalog_v1beta1/types
+   datacatalog_v1beta1/types
```
Comment (meredithslota) [inline docs/index.rst]: Owlbot keeps removing this. This commit tried to add it back but possibly I need to add it back somewhere else so it doesn't get overwritten? @tswast WWYD?

File: docs/index.rst [hunk 2]
```diff
@@ -23,17 +23,6 @@ API Reference
     datacatalog_v1beta1/types
 
 
-Migration Guide
```
Comment (meredithslota) [inline docs/index.rst:26]: Owlbot keeps removing this section. I tried to add it back in but Owlbot just removed it. Possibly I need to make these changes elsewhere? cc: @tswast WWYD?

Comment (parthea) [reply docs/index.rst]: I've pushed a commit to add `docs/UPGRADING.md` and the owlbot changes look good now. [Here](https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/common.py#L262) is the relevant synthtool code.

**Review by parthea — APPROVED** [review]

**Pull request merged**